src/event/writer.rs

```diff
@@ -161,9 +172,39 @@ impl EventWriter {
                 .expect("send error");
             rx_error
         } else {
+            self.event_handles.push_back(rx_flush);
```

Comment:
We need to support the case where the caller never calls flush without leaking memory. Currently the flush call removes items from `event_handles` but if it is not called this code will OOM.

Reply:
Do we need to periodically call flush to clear this? Or upon every write, call flush on all previous writes? Or clean up event_handles while doing every write? (It would need to go through every write call to check if its processed or not)

Reply:
Yes. On a write call, clear out any completed ones stopping at the first uncompleted one.